pigar/reqs.py

```diff
@@ -25,13 +25,19 @@ def project_import_modules(project_path, ignores):
     modules = ImportedModules()
     local_mods = list()
     cur_dir = os.getcwd()
+    ignore_paths = collections.defaultdict(set)
     if not ignores:
-        ignores = [os.path.join(project_path, d) for d in ['.git']]
+        ignore_paths[project_path] = {'.git'}
+    else:
+        for path in ignores:
+            parent_dir = os.path.dirname(path)
+            ignore_paths[parent_dir].add(os.path.basename(path))
 
     logger.info('Extracting project: {0}'.format(project_path))
     for dirpath, dirnames, files in os.walk(project_path, followlinks=True):
-        if dirpath.startswith(tuple(ignores)):
-            continue
+        if dirpath in ignore_paths:
+            dirnames[:] = [d for d in dirnames \
+                    if d not in ignore_paths[dirpath]]
```

Comment:
You really know what you are doing?

[os.walk](https://docs.python.org/2/library/os.html#os.walk)
> Generate the file names in a directory tree by walking the tree either top-down or bottom-up. For each directory in the tree rooted at directory top (including top itself), it yields a 3-tuple (dirpath, dirnames, filenames).
> dirpath is a string, the path to the directory. dirnames is a list of the names of the subdirectories in dirpath (excluding '.' and '..'). 

Also, about the code style: [PEP 0008](https://www.python.org/dev/peps/pep-0008/)